src/main/java/net/neoforged/neoforge/transfer/fluids/wrappers/BlockFluidHandler.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.fluids.wrappers;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.BucketPickup;
+import net.minecraft.world.level.block.LiquidBlockContainer;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.FluidState;
+import net.neoforged.neoforge.fluids.FluidType;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.fluids.FluidResource;
+import net.neoforged.neoforge.transfer.handlers.ISingleResourceHandler;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A handler for placing and picking up fluid blocks in the world.
+ */
+public class BlockFluidHandler implements ISingleResourceHandler<FluidResource> {
+    private static final Logger LOGGER = LogManager.getLogger();
```

Comment:
Is this the right place for this? I would assume `NeoForgeMod.LOGGER` would be the correct one to use, but if the standard is to have a unique one here, then disregard :)